### PR TITLE
fixed path custom library in dmh command

### PR DESCRIPTION
**Pull request by soez**
fixed when the custom library (LD_PRELOAD=..) matches the path of the native library, radare tries to find the hash of the debug library, something you will not find and it results seg fault

### CODE REVIEW

File: libr/core/linux_heap_glibc.c [hunk 1]
```diff
@@ -231,43 +231,38 @@ static bool GH(r_resolve_symbol)(RCore *core, GHT *symbol, const char *symname)
...
 	is_debug_file[2] = str_start_with (libc_ver_end, "/usr/lib64/");
 	is_debug_file[3] = str_start_with (libc_ver_end, "/lib/");
 	is_debug_file[4] = str_start_with (libc_ver_end, "/lib32/");
 	is_debug_file[5] = str_start_with (libc_ver_end, "/lib64/");
 
-	if (!is_debug_file[0] && !is_debug_file[1] && \
-	!is_debug_file[2] && !is_debug_file[3] && \
-	!is_debug_file[4] && !is_debug_file[5]) {
-		path = r_cons_input ("Is a custom library? (LD_PRELOAD=..) Enter full path glibc: ");
+	if (getenv("LD_PRELOAD")) {
```
Comment (radare) [inline libr/core/linux_heap_glibc.c]: Use r_sys_getenv and add a soace before the (

File: libr/core/linux_heap_glibc.c [hunk 2]
```diff
@@ -231,43 +231,38 @@ static bool GH(r_resolve_symbol)(RCore *core, GHT *symbol, const char *symname)
...
 		path = r_str_newf ("%s", libc_ver_end);
 		if (r_file_exists (path)) {
 			goto found;
 		}
 	}
-
 	if ((is_debug_file[3] || is_debug_file[4] || is_debug_file[5]) && \
 	r_file_is_directory ("/usr/lib/debug")) {
-		free (path);
 		path = r_str_newf ("%s%s", dir_dbg, libc_ver_end);
```
Comment (radare) [inline libr/core/linux_heap_glibc.c]: Leak here. Path is allocated and reassigned

Comment (soez) [reply libr/core/linux_heap_glibc.c]: done

File: libr/core/linux_heap_glibc.c [hunk 3]
```diff
@@ -231,33 +231,28 @@ static bool GH(r_resolve_symbol)(RCore *core, GHT *symbol, const char *symname)
...
 	is_debug_file[2] = str_start_with (libc_ver_end, "/usr/lib64/");
 	is_debug_file[3] = str_start_with (libc_ver_end, "/lib/");
 	is_debug_file[4] = str_start_with (libc_ver_end, "/lib32/");
 	is_debug_file[5] = str_start_with (libc_ver_end, "/lib64/");
 
-	if (!is_debug_file[0] && !is_debug_file[1] && \
-	!is_debug_file[2] && !is_debug_file[3] && \
-	!is_debug_file[4] && !is_debug_file[5]) {
-		path = r_cons_input ("Is a custom library? (LD_PRELOAD=..) Enter full path glibc: ");
+	if (r_sys_getenv ("LD_PRELOAD")) {
```
Comment (radare) [inline libr/core/linux_heap_glibc.c]: getenv returns a char*, so this line is leaking

File: libr/core/linux_heap_glibc.c (hunk 2 above)
Comment (radare) [reply libr/core/linux_heap_glibc.c]: uh? i still see the path is not freed before the r_str_newf

File: libr/core/linux_heap_glibc.c [hunk 4]
```diff
@@ -267,12 +262,10 @@ static bool GH(r_resolve_symbol)(RCore *core, GHT *symbol, const char *symname)
 			goto found;
 		}
 	}
-
 	if ((is_debug_file[3] || is_debug_file[4] || is_debug_file[5]) && \
 	r_file_is_directory ("/usr/lib/debug/.build-id")) {
 		get_hash_debug_file (libc_ver_end, hash, sizeof (hash) - 1);
 		libc_ver_end = hash;
-		free (path);
```
Comment (radare) [inline libr/core/linux_heap_glibc.c]: same here, why did you removed it?

**Pull request closed without merging**